tests/core/test_processor.py

```diff
@@ -241,15 +251,198 @@
     assert len((await jobs.scheduler()).get_jobs()) == 1
 
     # execute the jobs
-    await asyncio.sleep(5)
+    await wait_for_all_jobs(timeout_after_seconds=5.0)
```

Comment:
If a `TimeoutError` exception is raised here, should we remove all jobs?

Reply:
Raising the error lets the test fail. Don't those jobs get killed once the test fails?

Reply:
I don't think so, it doesn't look like we're doing that explicitly. Maybe we could add a fixture that stops all jobs after the test is run? I can't remember now though if `async` fixtures are allowed though.

Reply:
But I added `jobs.kill_scheduler()` now.

Reply:
Sorry, your message was not displayed until now.  Do you think the `jobs.kill_scheduler()` is enough? We haven't checked for uncanceled jobs before.